src/main/java/me/coley/recaf/util/self/SelfDependencyPatcher.java

```diff
@@ -29,10 +29,10 @@
 public class SelfDependencyPatcher {
 	private static final Path DEPENDENCIES_DIR_PATH = Recaf.getDirectory("dependencies");
 	private static final String[] DEPENDENCIES = new String[] {
-		"https://repo1.maven.org/maven2/org/openjfx/javafx-media/11.0.2/javafx-media-11.0.2-%s.jar",
-		"https://repo1.maven.org/maven2/org/openjfx/javafx-controls/11.0.2/javafx-controls-11.0.2-%s.jar",
-		"https://repo1.maven.org/maven2/org/openjfx/javafx-graphics/11.0.2/javafx-graphics-11.0.2-%s.jar",
-		"https://repo1.maven.org/maven2/org/openjfx/javafx-base/11.0.2/javafx-base-11.0.2-%s.jar"
+		"https://repo1.maven.org/maven2/org/openjfx/javafx-media/%d.0.2/javafx-media-%d.0.2-%s.jar",
+		"https://repo1.maven.org/maven2/org/openjfx/javafx-controls/%d.0.2/javafx-controls-%d.0.2-%s.jar",
+		"https://repo1.maven.org/maven2/org/openjfx/javafx-graphics/%d.0.2/javafx-graphics-%d.0.2-%s.jar",
+		"https://repo1.maven.org/maven2/org/openjfx/javafx-base/%d.0.2/javafx-base-%d.0.2-%s.jar"
```

Comment:
This works for major versions assuming the release is `X.0.0` but more work will need to be done if we want to target patch versions. Consider if a user is on JDK 15 and `15.0.0` is out but not the patch. 

I used these versions because its a fair lowest-common-denominator. JDK 11 is the version you should jump to after JDK 8 if you're being conservative. If the bytecode version of the current JVM is compatible with the latest library version there is no reason to not use a newer version of the library. I've not checked what bytecode level the newer library versions target.

My initial thoughts for doing this are:

1. Breaking this up into a series of sub-arrays with descending versions, grabbing the highest supported one.
2. Create a basic web scraper for the JavaFX maven directory and pull the latest version allowed.